front/pages/api/v1/w/[wId]/assistant/agent_configurations/search.ts

```diff
@@ -0,0 +1,124 @@
+import type { WithAPIErrorResponse } from "@dust-tt/types";
+import type { NextApiRequest, NextApiResponse } from "next";
+
+import { searchAgentConfigurationsByName } from "@app/lib/api/assistant/configuration";
+import { Authenticator, getAPIKey } from "@app/lib/auth";
+import type { AgentConfiguration } from "@app/lib/models/assistant/agent";
+import { apiError, withLogging } from "@app/logger/withlogging";
+
+/**
+ * @swagger
+ * /api/v1/w/{wId}/assistant/agent_configurations/search:
+ *   get:
+ *     summary: Search agent configurations by name
+ *     description: Search for agent configurations by name in the workspace identified by {wId}.
+ *     tags:
+ *       - Assistant
+ *     parameters:
+ *       - in: path
+ *         name: wId
+ *         required: true
+ *         description: ID of the workspace
+ *         schema:
+ *           type: string
+ *       - in: query
+ *         name: q
+ *         required: true
+ *         description: Search query for agent configuration names
+ *         schema:
+ *           type: string
+ *     security:
+ *       - BearerAuth: []
+ *     responses:
+ *       200:
+ *         description: Successfully retrieved agent configurations
+ *         content:
+ *           application/json:
+ *             schema:
+ *               type: object
+ *               properties:
+ *                 agentConfigurations:
+ *                   type: array
+ *                   items:
+ *                     $ref: '#/components/schemas/AgentConfiguration'
+ *       400:
+ *         description: Bad Request. Invalid or missing parameters.
+ *       401:
+ *         description: Unauthorized. Invalid or missing authentication token.
+ *       404:
+ *         description: Workspace not found.
+ *       405:
+ *         description: Method not supported. Only GET is expected.
+ *       500:
+ *         description: Internal Server Error.
+ */
+
+type GetAgentConfigurationsResponseBody = {
+  agentConfigurations: AgentConfiguration[];
+};
+
+async function handler(
+  req: NextApiRequest,
+  res: NextApiResponse<WithAPIErrorResponse<GetAgentConfigurationsResponseBody>>
+): Promise<void> {
+  const keyRes = await getAPIKey(req);
+  if (keyRes.isErr()) {
+    return apiError(req, res, keyRes.error);
+  }
+
+  const { wId, q } = req.query;
+  if (typeof wId !== "string") {
+    return apiError(req, res, {
+      status_code: 400,
+      api_error: {
+        type: "invalid_request_error",
+        message: "Workspace ID not found",
+      },
+    });
+  }
+
+  const { workspaceAuth } = await Authenticator.fromKey(keyRes.value, wId);
```

Comment:
You do need to check isBuilder (or at least isUSer) otherwise anyone with an API key can tap into that endpoint

Reply:
Hmm, Authenticator.fromKey() should already check that the key belongs to the workspace, right? 
Because as it's a public endpoint it will likely be used by API keys so not specifically linked to a user

Reply:
Nope it does not. It gives you a auth on the workspace (of the URL) and an auth on the workspace of the key.

But that does not mean that the auth has permission esp if the key workspace is not the same as the endpoint workspace. This is used for running dust apps from workspace Y as workspace X

Reply:
This is obviously not ideal yet and very dangerous

Reply:
👍 Understood